File: spring/src/main/java/cardsproject/domain/marketplace/OrderAuditLog.java
package cardsproject.domain.marketplace;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "orders_audit_log")
public class OrderAuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "record_id", nullable = false)
    private Long recordId;

    @Column(name = "field", nullable = false, length = 100)
    private String field;

    @Column(name = "old_value", columnDefinition = "TEXT")
    private String oldValue;

    @Column(name = "new_value", columnDefinition = "TEXT")
    private String newValue;

    @Column(name = "changed_at")
    private LocalDateTime changedAt = LocalDateTime.now();

    // getters/setters omitted — use Lombok @Data or generate in IDE
}

File: dotnet/CardsProject/Domain/Marketplace/OrderAuditLog.cs
namespace CardsProject.Domain.Marketplace;

public class OrderAuditLog
{
    public int Id { get; set; }
    public int RecordId { get; set; }
    public string Field { get; set; } = string.Empty;
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
    public DateTime ChangedAt { get; set; } = DateTime.UtcNow;
}
